src/zui/ZUITextEditor/helpers.ts

```diff
@@ -225,6 +227,23 @@ const convertSlateToRemarked = (
   return convertedChildren;
 };
 
+const shouldBeRemoved = (node: NodeEntry<Ancestor>): boolean => {
+  if (node && Object.prototype.hasOwnProperty.call(node[0], 'children')) {
+    if (
+      'children' in node[0].children[0] &&
+      Object.prototype.hasOwnProperty.call(node[0].children[0], 'children')
+    ) {
```

Comment:
This seems unnecessarily complex to me. I realize you probably found this solution in the Slate issue tracker, but I'm curious if you know the reason for using `hasOwnProperty.call(node[0]…` instead of `node[0]?.hasOwnProperty(…`, etc?

Reply:
Yes, the reason I was implemented so complex it's because <Ancestor> and the following types are a mixed of types. The reason to use `hasOwnProperty.call` it's to avoid a lint error. If I use `node[0]?.hasOwnProperty(…,` I get the error ` Do not access Object.prototype method 'hasOwnProperty' from target object`.